lib/internal/crypto/hash.js

```diff
@@ -188,8 +192,23 @@ async function asyncDigest(algorithm, data) {
   throw lazyDOMException('Unrecognized algorithm name', 'NotSupportedError');
 }
 
+function hash(algorithm, input, outputEncoding = 'hex') {
+  validateString(algorithm, 'algorithm');
+  if (typeof input !== 'string') {
+    validateBuffer(input, 'input');
+  }
+  // Fast case: if it's 'hex', we don't need to validate it further.
+  if (outputEncoding !== 'hex') {
+    validateString(outputEncoding);
+    outputEncoding = normalizeEncoding(outputEncoding) || outputEncoding;
```

Comment:
Just a minor thing, from what I saw, `normalizeEncoding` will return `undefined` for any invalid encoding, so maybe we could throw the error in the JS land instead of calling C++ and then throw the error.